### ISSUE TITLE
Add accessibility settings section to options

### ISSUE THREAD

**Issue by karmatosed** [WordPress/gutenberg#16354]
As settings like labels on the toolbar are added, it could be great to bring in a setting to options for accessibility. This could also include a note about reducing motion, potentially linking to a help document such as https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion#User_Preferences.

<img width="393" alt="options" src="https://user-images.githubusercontent.com/253067/60364323-bf6e2680-99dd-11e9-922f-34966bdb47d2.png">

A few things to get feedback on:

- What do you think of this option?
- What copy should be used for the information message on reducing motion?
- Should we link to a help document?
- Is there a better resource to link to if we do?

**Comment by kjellr** [comment]
> What copy should be used for the information message on reducing motion?

For what it's worth, I bet we could relatively easily tie this to a user setting as well. (If for instance, someone wanted to turn off animations in Gutenberg, but keep them on for their actual OS).

**Comment by mrwweb** [comment]
> What do you think of this option?

Three quick thoughts:
1. Icon + Text buttons are considered good for overall usability. Given that, I'd love to see them introduced with the option enabled by default (at least for new users/sites).
1. I think calling that section "Accessibility" pigeon-holes those options when they are actually useful to people for all sorts of reasons very much including personal preference. Like I said, it's good usability, not just accessibility. As for where to put it, this specific option feels very much aligned with other UI preferences like Top Toolbar, Spotlight Mode, and Fullscreen Mode.
1. I worry slightly that introducing a new Accessibility set of options could encourage future options that require people who rely on assistive technology or other accommodations to opt-in for a usable experience. On its own, this mockup doesn't convince me of the need for a brand new section.

> For what it's worth, I bet we could relatively easily tie this to a user setting as well. (If for instance, someone wanted to turn off animations in Gutenberg, but keep them on for their actual OS).

Many Windows users have accidentally turned on `prefers-reduced-motion` which suggests to me that a stand-alone option separate from the OS might be useful.

**Comment by aristath** [comment]
Just my 2c here: I don't think that accessibility should be opt-in. 
It should be available out of the box for everyone. 
If we need to provide options for design reasons (labels everywhere sometimes don't look cool) then it should be opt-out instead, though tbh I don't think there should be an option for it at all. As mentioned above, adding options for accessibility will just encourage bad decisions and a way of thinking like "we'll do what's pretty and then dumb it down for those that need it". It sets a bad precedent.

**Comment by kjellr** [comment]
For now, I think we could just place this option under the "General Settings" header: there are only two other options there, and I don't think this would feel out of place. If and when that section grows, or when we have more similar options to group together, we can break it out into a new settings group. 

Regarding the default state of the labels — that feels likes something to discuss in #10524 instead. Either way, a checkbox to turn those on and off will be necessary.

**Comment by melchoyce** [comment]
> For now, I think we could just place this option under the "General Settings" header

Let's do this.

**Comment by tomjn** [comment]
Just noting that animations are broken on Firefox when resist fingerprinting is turned on. There doesn't appear to be a way to check for this. Also, not all browsers support the motion reduction parameter, or pass on that information to JS

**Comment by Wingo5315** [comment]
@tomjn I think that, until we figure out how to find out if resist fingerprinting is on, we should disable all animations in Firefox. Or we could have a dialog for Firefox users that says, "Allow fingerprinting for this website to get the best experience."

**Comment by karmatosed** [comment]
I am going to close this as a lot of time has passed and discussion is going on in other places around this, which is more focused.
